gencode.c

```diff
@@ -1552,7 +1585,7 @@ gen_load_macplrel(offset, size)
 		 * as an offset.
 		 */
 		s2 = new_stmt(BPF_LD|BPF_IND|size);
-		s2->s.k = offset;
+		s2->s.k = off_macpl + offset;
```

Comment:
`off_macpl` is the offset of the MAC-layer *payload*, and points *past* the link-layer header, so presumably this should also be a different variable.